### PR TITLE
Made small readability improvements.

### PR DESCRIPTION
**Pull request by MartinThoma**
The code should do the same thing, but hopefully is easier to read

### CODE REVIEW

File: django/utils/cache.py [hunk 1]
```diff
@@ -181,10 +181,13 @@ def get_conditional_response(request, etag=None, last_modified=None, response=No
...
     # Step 4: Test the If-Modified-Since precondition.
-    if (not if_none_match_etags and if_modified_since and
-            not _if_modified_since_passes(last_modified, if_modified_since)):
-        if request.method in ('GET', 'HEAD'):
-            return _not_modified(request, response)
+    if (
+        not if_none_match_etags and
+        if_modified_since and
+        not _if_modified_since_passes(last_modified, if_modified_since) and
+        request.method in ('GET', 'HEAD'):
```
Comment (pope1ni) [inline django/utils/cache.py:188]:
```suggestion
        request.method in ('GET', 'HEAD')
```

**Review by pope1ni — CHANGES_REQUESTED** [review]
Please can you revise and ensure there are no other syntax errors?

**Comment by MartinThoma** [comment]
I've just rebased.

**Comment by MartinThoma** [comment]
I was also wondering about this: https://github.com/django/django/blob/master/django/core/serializers/base.py#L117 (`self.first = self.first and False`)

File: django/db/models/query.py [hunk 2]
```diff
@@ -1228,15 +1228,12 @@ def ordered(self):
             return True
         if self.query.extra_order_by or self.query.order_by:
             return True
-        elif (
+        return (
```
Comment (pope1ni) [inline django/db/models/query.py:1231]: Because of Python's support for [short-circuiting](https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not) boolean operators, the tests are failing here as sometimes `[]` is being returned. The fix is easy:

```suggestion
        return bool(
```

**Review by pope1ni — CHANGES_REQUESTED** [review]
Please can you fully run the test suite locally and ensure that all issues are resolved.

You can find information on [running Django's tests](https://docs.djangoproject.com/en/3.1/internals/contributing/writing-code/unit-tests/) in the documentation.

File: django/utils/cache.py [hunk 3]
```diff
@@ -180,10 +180,13 @@ def get_conditional_response(request, etag=None, last_modified=None, response=No
             return _precondition_failed(request)
 
     # Step 4: Test the If-Modified-Since precondition.
-    if (not if_none_match_etags and if_modified_since and
-            not _if_modified_since_passes(last_modified, if_modified_since)):
-        if request.method in ('GET', 'HEAD'):
-            return _not_modified(request, response)
+    if (
+        not if_none_match_etags and
+        if_modified_since and
```
Comment (jacobtylerwalls) [inline django/utils/cache.py:185]: PEP 8 hedges about line breaks and binary operators but suggests ultimately that breaking before the operator is better. As long as we're touching this I would suggest breaking before the operators.

File: django/db/models/query.py [hunk 4]
```diff
@@ -1228,15 +1228,12 @@ def ordered(self):
...
+        return bool(
             self.query.default_ordering and
             self.query.get_meta().ordering and
             # A default ordering doesn't affect GROUP BY queries.
             not self.query.group_by
-        ):
-            return True
-        else:
-            return False
+        )
```
Comment (felixxm) [inline django/db/models/query.py:1236]: I don't think it's an improvement, it's less readable IMO.

File: django/shortcuts.py [hunk 5]
```diff
@@ -121,10 +121,9 @@ def resolve_url(to, *args, **kwargs):
         # further to some Python functions like urlparse.
         to = str(to)
 
-    if isinstance(to, str):
+    if isinstance(to, str) and to.startswith(('./', '../')):
         # Handle relative URLs
-        if to.startswith(('./', '../')):
-            return to
+        return to
```
Comment (felixxm) [inline django/shortcuts.py:126]: Above comment doesn't match to the line anymore.

File: django/contrib/staticfiles/finders.py [hunk 6]
```diff
@@ -187,12 +187,11 @@ def find_in_app(self, app, path):
...
         """
         storage = self.storages.get(app)
-        if storage:
+        if storage and storage.exists(path):
             # only try to find a file if the source dir actually exists
-            if storage.exists(path):
-                matched_path = storage.path(path)
-                if matched_path:
-                    return matched_path
+            matched_path = storage.path(path)
```
Comment (felixxm) [inline django/contrib/staticfiles/finders.py:192]: Above comment doesn't match to the line anymore.

File: django/db/models/fields/__init__.py [hunk 7]
```diff
@@ -782,12 +782,11 @@ def contribute_to_class(self, cls, name, private_only=False):
...
         self.model = cls
         cls._meta.add_field(self, private=private_only)
-        if self.column:
+        if self.column and not getattr(cls, self.attname, None):
             # Don't override classmethods with the descriptor. This means that
             # if you have a classmethod and a field with the same name, then
             # such fields can't be deferred (we don't have a check for this).
-            if not getattr(cls, self.attname, None):
-                setattr(cls, self.attname, self.descriptor_class(self))
+            setattr(cls, self.attname, self.descriptor_class(self))
```
Comment (felixxm) [inline django/db/models/fields/__init__.py:789]: Above comment doesn't match to the line anymore. Moreover this code is really complicated and fragile, please revert this change.

File: django/db/backends/sqlite3/operations.py [hunk 8]
```diff
@@ -282,15 +282,13 @@ def convert_datetimefield_value(self, value, expression, connection):
...
+        if value is not None and not isinstance(value, datetime.date):
+            value = parse_date(value)
         return value
 
     def convert_timefield_value(self, value, expression, connection):
-        if value is not None:
-            if not isinstance(value, datetime.time):
-                value = parse_time(value)
+        if value is not None and not isinstance(value, datetime.time):
+            value = parse_time(value)
```
Comment (felixxm) [inline django/db/backends/sqlite3/operations.py:291]: Please revert changes in converters, these conditions are prepare for handling different types of values. I don't see any value in changing this.

File: django/test/html.py [hunk 9]
```diff
@@ -23,29 +23,25 @@ def __init__(self, name, attributes):
...
+                self.children[-1] = normalize_whitespace(self.children[-1])
+                return
+        elif self.children and isinstance(self.children[-1], str) and self.children[-1].isspace():
             # removing last children if it is only whitespace
             # this can result in incorrect dom representations since
             # whitespace between inline tags like <span> is significant
-            if isinstance(self.children[-1], str):
-                if self.children[-1].isspace():
-                    self.children.pop()
+            self.children.pop()
```
Comment (felixxm) [inline django/test/html.py:34]: Above comment doesn't match to the line anymore.

File: django/utils/cache.py (hunk 3 above)
Comment (felixxm) [reply django/utils/cache.py:185]: In Django we break after the operator, that's why `W504` is ignored. It's checked by `flake8`.

**Review by felixxm — COMMENTED** [review]
@MartinThoma Thanks for this patch :+1: However, personally, I don't think that most of these changes improve readability and they will create unnecessary noise in the history. I'm not sure about them, :heavy_minus_sign: / :heavy_plus_sign: .

**Comment by MartinThoma** [comment]
Thank you for having a look at it and the detailed feedback! I will improve the PR once I have the time (unlikely to happen this week)

**Comment by felixxm** [comment]
@MartinThoma Thanks :+1: I pushed small edits.
